File: basics/counter/seahorse/programs_py/counter_seahorse.py
# counter_seahorse
# Built with Seahorse v0.1.6

from seahorse.prelude import *

declare_id('Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS')


class Counter(Account):
    count: u64


@instruction
def initialize_counter(counter: Empty[Counter], payer: Signer, seed: u8):
    counter.init(
        payer=payer,
        seeds=[seed]
    )


@instruction
def increment(counter: Counter):
    counter.count += 1

File: basics/counter/seahorse/tests/counter_seahorse.ts
import * as anchor from '@coral-xyz/anchor';
import type { Program } from '@coral-xyz/anchor';
import { Keypair, PublicKey, SystemProgram } from '@solana/web3.js';
import { assert } from 'chai';
import type { CounterSeahorse } from '../target/types/counter_seahorse';

describe('counter_seahorse', () => {
  // Configure the client to use the local cluster.
  anchor.setProvider(anchor.AnchorProvider.env());

  const program = anchor.workspace.CounterSeahorse as Program<CounterSeahorse>;

  it('Increment counter', async () => {
    const seed = 69;
    const counter = PublicKey.findProgramAddressSync([Buffer.from([0x45])], program.programId)[0];

    // Initialize counter
    await program.methods
      .initializeCounter(seed)
      .accounts({
        payer: program.provider.publicKey,
      })
      .rpc();

    // Increment counter
    await program.methods
      .increment()
      .accounts({
        counter,
      })
      .rpc();

    const count = (await program.account.counter.fetch(counter)).count.toNumber();
    assert(count === 1, 'Expected count to be 1');
  });
});
